embassy-extras/src/peripheral.rs

```diff
@@ -52,22 +157,61 @@ impl<S: PeripheralState> PeripheralMutex<S> {
         this.irq_setup_done = true;
     }
 
-    pub fn with<R>(self: Pin<&mut Self>, f: impl FnOnce(&mut S, &mut S::Interrupt) -> R) -> R {
+    pub fn with<R>(self: Pin<&mut Self>, f: impl FnOnce(&mut S) -> R) -> R {
         let this = unsafe { self.get_unchecked_mut() };
 
+        let was_enabled = this.irq.is_enabled();
```

Comment:
It should be fine to assume the interrupt is permanently enabled for the entire existence of the PeripheralMutex. No existing driver needs dynamically enabling/disabling the interrupt, peripherals already have their `IE` flags.

This would allow removing this check, which is nice because this code is rather hot.